app/models/owner.rb

```diff
@@ -9,22 +9,23 @@ class Owner < ApplicationRecord
          :confirmable, :recoverable, jwt_revocation_strategy: JwtDenylist
 
   validates :email, uniqueness: true, presence: true
-  validates :password, 
-    presence: true, 
-    length: { in: Devise.password_length }, 
-    confirmation: true, 
-    on: :create 
-
-  validates :password, 
-    allow_nil: true, 
-    length: { in: Devise.password_length }, 
-    confirmation: true, 
+  validates :password,
+    presence: true,
+    length: { in: Devise.password_length },
+    confirmation: true,
+    on: :create
+
+  validates :password,
+    allow_nil: true,
+    length: { in: Devise.password_length },
+    confirmation: true,
     on: :update
 
```

Comment:
Right now, when migrating, we are going through three stages:

 1. The app as is right now, `owner.affiliate` will return a string
 2. The apps db is migrated, but the data migration script has not been run yet. Since the code is updated, the app should expect `.affiliate` to return a model, but it's always empty. If payments are executed in this period, owners with affiliates might pay the full price, or similar weird behaviour might occur
 3. The data migration has run, now `.affilaite` will return the affilaite model
 
So I think for the transition it might make sense to overwrite the `owner.affiliate` method so it always returns the affiliate model. Something like

```suggestion

  def affiliate
    return super if super.present? # make sure the association is used after the data migration
    
    Affiliate.find_by(code: affiliate_code) # before the data migration is run, we try to find the affiliate "the old fashioned way"
  end
```